packages/@aws-cdk/aws-appmesh/lib/private/utils.ts

```diff
@@ -125,10 +126,19 @@ export function validateGrpcMatchArrayLength(metadata?: HeaderMatch[]): void {
 }
 
 /**
- * This is the helper method to validate at least one of gRPC match type is defined.
+ * This is the helper method to validate at least one of gRPC route match type is defined.
  */
-export function validateGrpcMatch(match: GrpcRouteMatch): void {
+export function validateGrpcRouteMatch(match: GrpcRouteMatch): void {
   if (match.serviceName === undefined && match.metadata === undefined && match.methodName === undefined) {
     throw new Error('At least one gRPC match property must be provided');
   }
 }
+
+/**
+ * This is the helper method to validate at least one of gRPC gateway route match type is defined.
+ */
+export function validateGrpcGatewayRouteMatch(match: GrpcGatewayRouteMatch): void {
+  if (match.serviceName === undefined && match.metadata === undefined && match.hostname === undefined) {
+    throw new Error('At least one gRPC gateway route match property must be provided');
```

Comment:
I think this is not a great message, since `GrpcGatewayRouteMatch` also contains `rewriteRequestHostname`.

I think you need to mention in the error message the set of properties from which one is required.

Reply:
@skinny85 - Great point. I will call out as `beside rewriteRequestHostname` to reduce the line. What do you think?